lib/pure/asyncstreams.nim

```diff
@@ -107,7 +121,10 @@ proc read*[T](future: FutureStream[T]): owned(Future[(bool, T)]) =
       res[0] = true
       res[1] = fs.queue.popFirst()
```

Comment:
I think it might be better to avoid calling this if `fs` has failed, instead return early from this callback and fail `resFut` before this branch. I think there is a possibility this line can throw an exception otherwise.

Reply:
We're not calling this if `fs` has failed, as failed also means `finished`, and we don't get into this branch.